### PR TITLE
Family heirloom rework/refactor

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Continued from #5860
This PR has been made again because I can't open the original one.
also, I don't know how to make the commit histories look better...

Note: define changes should happen at another PR since it touches a lot of job strings, but I'd like to focus on the family heirloom, not the change that touches a lot of codes which can cause some bug.

## Why It's Good For The Game
Family Heirloom is now more gimmicky.
I tested myself a bit, taking 'kitty ears' family heirloom as a human was quite awkward. lol

## Things to take a look
It looks the item table needs to have more diversity. Since this is a test version, adding 'interesting' items will be progressed one by one.
* which items will be added more?
* should mime/clown get only the expected item?

## Changelog
:cl:
refactor: Refactored how Family Heirloom trait works in the code to make the system more flexible.
tweak: Family Heirloom trait now grants an item from a wider variety of the table based on your species, job, department - including a rare chance to get some suspicious looking items.
/:cl:

### CODE REVIEW

**Comment by MNarath1** [comment]
If you want to tidy up your commit history you can squash commits together trough usually as long as all commits are related to the PR its usually not needed

**Comment by MNarath1** [comment]
Also how tf is the mapdiff bot failing here

**Comment by MNarath1** [comment]
Ah wait merge conflicts right

**Comment by EvilDragonfiend** [comment]
> If you want to tidy up your commit history you can squash commits together trough usually as long as all commits are related to the PR its usually not needed

Thanks. I'll take a look on the feature.

> Ah wait merge conflicts right

I have no idea why the merge conflict happens

**Comment by MNarath1** [comment]
> > If you want to tidy up your commit history you can squash commits together trough usually as long as all commits are related to the PR its usually not needed
> 
> Thanks. I'll take a look on the feature.
> 
> > Ah wait merge conflicts right
> 
> I have no idea why the merge conflict happens

try to merge the current beestation master into your branche it will show all conflicting files most likely someone modified one of the files you modified and it can't merge it automatically so you have to choose in those areas yourself what to merge in and what not (or in what order ) to fix the conflicts

File: code/datums/traits/negative.dm [hunk 1]
```diff
@@ -120,13 +120,13 @@
 
 /datum/quirk/foreigner/add()
 	var/mob/living/carbon/human/H = quirk_holder
-	if(ishuman(H) && H.job != JOB_CURATOR)
+	if(ishuman(H) && H.job != "Curator")
```
Comment (MNarath1) [inline code/datums/traits/negative.dm:123]: why are you changing this back to the string when you already haved defines for those strings in your branch

**Comment by MNarath1** [comment]
also for your owns sanity's sake i would recommend you to actually make branches for your PR's and not using master the master branch should usually stay unmodified and you keep it up to date by making the beestation master the upstream that way you can easily make new branches of your master for new PR's and you won'T have the old commits of other PR cogging it up too.

File: code/datums/traits/negative.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/datums/traits/negative.dm:123]: because I am going to make it at another PR that would make large change about job string defines

**Comment by EvilDragonfiend** [comment]
This was closed because I fucked up commits. There's another PR I made.

**Pull request closed without merging**